docs/ActiveLabel.rst

```diff
@@ -1,135 +1,320 @@
+      # instance of that class is created. We'll call this created object obj A.
+      # If obj A's class includes this ``ActiveLabel``, and, additionally, obj A has the label associated
+      # with this ``ActiveLabel``, then this included block will be evaluated
+      # within the context of obj A.
     end
 
     module InstanceMethods
-      def destroy
-        destroyed_at = Time.now
+      # After obj A has been found and initialized, before the included block is evaluated, obj A will
+      # be extended with these InstanceMethods (e.g. obj.extend(InstanceMethods))
 
-        super
+      def act
+        puts "I acted!"
       end
     end
 
     module ClassMethods
-      def destroyed_recently
-        all.where("#{identity}.destroyed_at > ?", 1.week.ago)
+      # Similar to ``ActiveSupport::Concern``, when this ``ActiveLabel`` module is included in an
+      # ``ActiveNode`` class, the class will be extended with these singleton methods (e.g. Person.extend(ClassMethods))
+
+      def actor_popularity_scale
+        puts "5 stars = excellent. 1 star = poor."
       end
     end
   end
 
+``ActiveLabel`` modules only describe functionality that is tied to a label. Actually adding that label to instances of a class
+is a seperate step. If you'd like to add a label to specific instances of a class, you can use standard ``neo4j-core`` methods
+``add_label()`` or ``remove_label``. You can also use special helper methods that ``ActiveLabel`` adds to a class when it is
+included in a class
 
-Creating
---------
+.. code-block:: ruby
+
+  # Initializes a Person with additional Actor label
+  Person.actor.new
+
+  # Creates a Person with additional Actor label
+  Person.actor.create
+
+  # Creates a Person with additional Actor AND Director labels
+  Person.actor.director.create
+
+If you'd like to `always` add one or more additional labels to instances of a class, you can use the ``ActiveNode`` ``label`` method
 
-If an ``ActiveLabel`` does not declare ``follows_label``, creating a node will attach the corresponding label.  Otherwise you must trigger the attachment of the label:
+.. code-block:: ruby
+
+  class Person
+    include Neo4j::ActiveNode
+    include Actor
+    include Director
+    include User
+
+    # ``label :Actor, optional: true`` automatically adds the label ``:Actor``
+    # to every instance of the Person class. The :Actor label is technically
+    # optional, even though it is always added, because a node will still be mapped
+    # to the Person class even if you manually remove the :Actor label from it.
+    label :Actor, optional: true
+
+    # If you call the ``label`` method without the ``optional: true`` argument,
+    # then nodes will only be mapped to the Person class if the label is
+    # also present on the node. (i.e. removing the :User label from a node will
+    # mean that that node is no longer considered a Person)
+    label :User
```

Comment:
Note, if the `Person` class has `label :User` (or `label :User, optional: true`), then there would be no difference between `Person.create` and `Person.user.create`. This should probably be explicitly mentioned.